pypeit/spec2dobj.py

```diff
@@ -415,22 +416,59 @@ def write_to_fits(self, outfile, pri_hdr=None, update_det=None, overwrite=True):
                 Header to be used in lieu of default
                 Usually generated by :func:`pypeit,spec2dobj.AllSpec2DObj.build_primary_hdr`
             update_det (list, optional):
-                Detector to be updated
+                Detector to be updated.  If given, all of the other
+                detectors remain as they were in the file (if the file exists)
+            slitspatnum (:obj:`str` or :obj:`list`, optional):
+              Restricted set of slits for reduction
+              If provided, do not clobber the existing file but only update
+              the indicated slits.  Useful for re-running on a subset of slits
             overwrite (bool, optional):
 
         """
         if os.path.isfile(outfile):
+            # Clobber?
             if not overwrite:
                 msgs.warn("File {} exits.  Use -o to overwrite.".format(outfile))
                 return
+            # Load up the original
+            _allspecobj = AllSpec2DObj.from_fits(outfile)
+            # Replace the newly reduced detector?
             if update_det is not None:
-                # Load up and replace
-                _allspecobj = AllSpec2DObj.from_fits(outfile)
                 for det in _allspecobj.detectors:
                     if det in np.atleast_1d(update_det):
                         continue
                     else:
                         self[det] = _allspecobj[det]
+            elif slitspatnum is not None: # Update specific slits!
+                
+                # Grab modified detectors and slits
+                dets, spat_ids = slittrace.parse_slitspatnum(slitspatnum)
```

Comment:
I think this got moved to `pypeit/core/parse.py` in `develop`.

Reply:
right you are!  fixed